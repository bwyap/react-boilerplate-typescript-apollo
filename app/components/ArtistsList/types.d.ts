import { FunctionalComponent } from '../../typings/component';
import { Artist } from '../../graphql/types';

export interface ArtistsListComponentFn
  extends FunctionalComponent<ArtistsListProps> {}

export interface ArtistsListProps {
  loading?: boolean;
  error?: any;
  artists?: Artist[];
}

export interface ArtistListItemComponentFn
  extends FunctionalComponent<ArtistListItemProps> {}

export interface ArtistListItemProps {
  item: Artist;
}
