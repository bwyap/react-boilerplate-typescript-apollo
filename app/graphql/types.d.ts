export interface Artist {
  id: string;
  name: string;
  image: string;
  albums: Array<{
    id: string;
    name: string;
  }>;
}
