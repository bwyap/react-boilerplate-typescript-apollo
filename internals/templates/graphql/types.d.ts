/**
 * Define types returned by your GraphQL operations here.
 */
