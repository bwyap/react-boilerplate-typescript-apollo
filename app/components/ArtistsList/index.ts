import ArtistsList from './parts/ArtistsList';

export default ArtistsList;
