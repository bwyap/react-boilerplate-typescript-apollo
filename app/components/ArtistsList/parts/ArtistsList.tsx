import * as React from 'react';
import * as PropTypes from 'prop-types';

import List from '../../../components/List';
import ListItem from '../../../components/ListItem';
import LoadingIndicator from '../../../components/LoadingIndicator';
import { ArtistsListComponentFn } from '../types';
import { ArtistListItem } from './ArtistListItem';

const ArtistsList: ArtistsListComponentFn = ({
  loading = false,
  error = null,
  artists = null,
}) => {
  if (loading) {
    return <List component={LoadingIndicator} />;
  }

  if (error) {
    const ErrorComponent = () => (
      <ListItem item="Something went wrong, please try again!" />
    );
    return <List component={ErrorComponent} />;
  }

  if (artists) {
    return <List items={artists} component={ArtistListItem} />;
  }

  return null;
};

ArtistsList.propTypes = {
  loading: PropTypes.bool,
  error: PropTypes.any,
  artists: PropTypes.arrayOf((ArtistListItem.propTypes as any).item),
};

export default ArtistsList;
