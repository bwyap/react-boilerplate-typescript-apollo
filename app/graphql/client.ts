import ApolloClient, { PresetConfig } from 'apollo-boost';

import loadEnvironment from '../env';
import { MyReduxStore } from '../typings/store';

/**
 * Create Apollo Client
 */
export default async (store: MyReduxStore) => {
  const env = await loadEnvironment();

  const config: PresetConfig = {
    uri: `${env.example}/graphql`,
    fetchOptions: {
      headers: {
        // Add authorization to each request here
        // authorization: store.getState().app.token,
      },
    },
  };

  return new ApolloClient(config);
};
