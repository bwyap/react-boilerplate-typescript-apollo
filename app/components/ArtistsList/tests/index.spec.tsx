import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import ArtistsList from '../index';
import { Artist } from '../../../graphql/types';

describe('<ReposList />', () => {
  it('should render the loading indicator when its loading', () => {
    const { container } = render(<ArtistsList loading />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an error if loading failed', () => {
    const { queryByText } = render(
      <IntlProvider locale="en">
        <ArtistsList loading={false} error={{ message: 'Loading failed!' }} />
      </IntlProvider>,
    );
    expect(queryByText(/Something went wrong/)).not.toBeNull();
  });

  it('should render the repositories if loading was successful', () => {
    const artists: Artist[] = [
      {
        id: '3AQRLZ9PuTAozP28Skbq8V',
        name: 'The Script',
        image:
          'https://i.scdn.co/image/15913ff470e612866f0f32179a225cacf546b4a7',
        albums: [
          {
            id: '3W9NClLDhTHyRmy8ZLfoJf',
            name: 'Freedom Child',
          },
        ],
      },
    ];
    const { container } = render(
      <IntlProvider locale="en">
        <ArtistsList artists={artists} error={null} />
      </IntlProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render anything if nothing interesting is provided', () => {
    const { container } = render(<ArtistsList error={null} loading={false} />);

    expect(container.firstChild).toBeNull();
  });
});
