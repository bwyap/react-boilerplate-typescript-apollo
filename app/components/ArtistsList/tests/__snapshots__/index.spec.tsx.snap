// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReposList /> should render the loading indicator when its loading 1`] = `
<div
  class="sc-bwzfXH bycCir"
>
  <ul
    class="sc-bdVaJa kjaCOS"
  >
    <div
      class="sc-ifAKCX jcAqoe"
      data-testid="loading"
    >
      <div
        class="sc-EHOje jorIkb"
      />
      <div
        class="sc-bZQynM cCfJFf"
      />
      <div
        class="sc-gzVnrw gnryHA"
      />
      <div
        class="sc-htoDjs jRhOQV"
      />
      <div
        class="sc-dnqmqq bVoquB"
      />
      <div
        class="sc-iwsKbI bEwLhw"
      />
      <div
        class="sc-gZMcBi gVTMMN"
      />
      <div
        class="sc-gqjmRU hqbrcO"
      />
      <div
        class="sc-VigVT bjQYpr"
      />
      <div
        class="sc-jTzLTM jAgAmf"
      />
      <div
        class="sc-fjdhpX gAXUfu"
      />
      <div
        class="sc-jzJRlG bsUthD"
      />
    </div>
  </ul>
</div>
`;

exports[`<ReposList /> should render the repositories if loading was successful 1`] = `
<div
  class="sc-bwzfXH bycCir"
>
  <ul
    class="sc-bdVaJa kjaCOS"
  >
    <li
      class="sc-bxivhb cHNIkg"
    >
      <div
        class="sc-htpNat cdjzok"
      >
        The Script
         (
        1
         album
        
        )
      </div>
    </li>
  </ul>
</div>
`;
